wgengine/magicsock/magicsock.go

```diff
@@ -3064,3 +3088,86 @@ func (le *lazyEndpoint) GetPeerEndpoint(peerPublicKey [32]byte) conn.Endpoint {
 	le.c.logf("magicsock: lazyEndpoint.GetPeerEndpoint(%v) found: %v", pubKey.ShortString(), ep.nodeAddr)
 	return ep
 }
+
+type Path string
+
+const (
+	PathDirectIPv4 Path = "direct_ipv4"
+	PathDirectIPv6 Path = "direct_ipv6"
+	PathDERP       Path = "derp"
+)
+
+type pathLabel struct {
+	// Path indicates the path that the packet took:
+	// - direct_ipv4
+	// - direct_ipv6
+	// - derp
+	Path Path
+}
+
+var (
+	metricInboundPacketsTotal = usermetric.NewMultiLabelMap[pathLabel](
+		"tailscaled_inbound_packets_total",
+		"counter",
+		"Counts the number of packets received from other peers",
+	)
+	metricOutboundPacketsTotal = usermetric.NewMultiLabelMap[pathLabel](
+		"tailscaled_outbound_packets_total",
+		"counter",
+		"Counts the number of packets sent to other peers",
+	)
+	metricInboundBytesTotal = usermetric.NewMultiLabelMap[pathLabel](
+		"tailscaled_inbound_bytes_total",
+		"counter",
+		"Counts the number of bytes received from other peers",
+	)
+	metricOutboundBytesTotal = usermetric.NewMultiLabelMap[pathLabel](
+		"tailscaled_outbound_bytes_total",
+		"counter",
+		"Counts the number of bytes sent to other peers",
+	)
+)
+
+func init() {
+	// Configure the expvar metrics to expose the current value of the clientmetrics
+	// instead of counting it twice.
+	metricInboundPacketsTotal.Set(pathLabel{Path: PathDERP}, expvar.Func(func() any {
+		return metricRecvDataPacketsDERP.Value()
+	}))
+	metricInboundPacketsTotal.Set(pathLabel{Path: PathDirectIPv4}, expvar.Func(func() any {
+		return metricRecvDataPacketsIPv4.Value()
+	}))
+	metricInboundPacketsTotal.Set(pathLabel{Path: PathDirectIPv6}, expvar.Func(func() any {
+		return metricRecvDataPacketsIPv6.Value()
+	}))
+
+	metricOutboundPacketsTotal.Set(pathLabel{Path: PathDERP}, expvar.Func(func() any {
+		return metricSendDataPacketsDERP.Value()
+	}))
+	metricOutboundPacketsTotal.Set(pathLabel{Path: PathDirectIPv4}, expvar.Func(func() any {
+		return metricSendDataPacketsIPv4.Value()
+	}))
+	metricOutboundPacketsTotal.Set(pathLabel{Path: PathDirectIPv6}, expvar.Func(func() any {
+		return metricSendDataPacketsIPv6.Value()
+	}))
+
+	metricInboundBytesTotal.Set(pathLabel{Path: PathDERP}, expvar.Func(func() any {
+		return metricRecvDataBytesDERP.Value()
+	}))
+	metricInboundBytesTotal.Set(pathLabel{Path: PathDirectIPv4}, expvar.Func(func() any {
+		return metricRecvDataBytesIPv4.Value()
+	}))
+	metricInboundBytesTotal.Set(pathLabel{Path: PathDirectIPv6}, expvar.Func(func() any {
+		return metricRecvDataBytesIPv6.Value()
+	}))
+
+	metricOutboundBytesTotal.Set(pathLabel{Path: PathDERP}, expvar.Func(func() any {
+		return metricSendDataBytesDERP.Value()
+	}))
+	metricOutboundBytesTotal.Set(pathLabel{Path: PathDirectIPv4}, expvar.Func(func() any {
+		return metricSendDataBytesIPv4.Value()
+	}))
+	metricOutboundBytesTotal.Set(pathLabel{Path: PathDirectIPv6}, expvar.Func(func() any {
+		return metricSendDataBytesIPv6.Value()
+	}))
+}
```

Comment:
this is all very repetitive looking. We can factor this out.

But also: boxing an int64 into an `any` interface allocates. I think we can avoid that. Doing it occasionally it bearable but doing it for all our metrics seems painful.